pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachinePciDevices/index.vue

```diff
@@ -0,0 +1,218 @@
+
+    // pciDevice is one per device per node - if multiple nodes have device or multiple devices on a node there will be duplicate deviceID vendorID
+    uniqueDevices() {
+      const out = {};
+
+      this.enabledDevices.forEach((deviceCRD) => {
+        const uniqueId = deviceCRD.uniqueId;
+        const deviceNode = deviceCRD?.status?.nodeName;
+
+        if (!out[uniqueId]) {
+          out[uniqueId] = {
+            nodes:      [deviceNode],
+            deviceCRDs: [deviceCRD]
+          };
+        } else {
+          if (!out[uniqueId].nodes.includes(deviceNode)) {
+            out[uniqueId].nodes.push(deviceNode);
+          }
+          out[uniqueId].deviceCRDs.push(deviceCRD);
+        }
+      });
+
+      return out;
+    },
+
+    devicesByNode() {
+      const out = {};
+
+      for (const deviceUid in this.uniqueDevices) {
+        const nodesWithDevice = this.uniqueDevices[deviceUid].nodes;
+
+        nodesWithDevice.forEach((node) => {
+          if (!out[node]) {
+            out[node] = [deviceUid];
+          } else {
+            out[node].push(deviceUid);
+          }
+        });
+      }
+
+      return out;
+    },
+
+    // determine which nodes contain all devices selected
+    compatibleNodes() {
+      const out = [...Object.keys(this.devicesByNode)];
+
+      this.selectedDevices.forEach((deviceUid) => {
+        remove(out, (nodeName) => {
+          const nodesWithDevice = this.uniqueDevices[deviceUid].nodes;
+
+          return !nodesWithDevice.includes(nodeName);
+        });
+      });
+
+      return out;
+    },
+
+    // format an array of available devices for the dropdown
+    deviceOpts() {
+      return Object.keys(this.uniqueDevices).map((deviceId) => {
+        const device = this.uniqueDevices[deviceId].deviceCRDs[0];
+
+        return {
+          resourceName: device?.status?.resourceName,
+          value:        deviceId,
+          label:        deviceId
+        };
+      });
+    },
+  },
+
+  methods: {
+    nodeNameFromUid(uid) {
+      for (const deviceUid in this.uniqueDevices) {
+        const nodes = this.uniqueDevices[deviceUid].nodes;
+        const thisNode = nodes.find(node => node.systemUUID === uid);
+
+        if (thisNode) {
+          return thisNode.name;
+        }
+      }
+    },
+  }
+};
+</script>
+
+<template>
+  <div>
+    <div class="row">
+      <div class="col span-12">
+        <Banner color="info">
+          <t k="harvester.pci.howToUseDevice" />
+        </Banner>
+      </div>
+    </div>
+    <template v-if="enabledDevices.length">
+      <div class="row">
+        <div class="col span-6">
+          <LabeledSelect
+            v-model="selectedDevices"
+            label="Available PCI Devices"
+            searchable
+            multiple
+            taggable
+            :options="deviceOpts"
+          >
+            <template #option="option">
+              <span>{{ option.value }} <span class="text-label">{{ option.resourceName }}</span></span>
```

Comment:
We could show `description` here and use "ugly" `resourceName` under the hood.

Reply:
@mantis-toboggan-md    PTAL， thanks